integrations/operator/embeddedtbot/syncclient/cache.go

```diff
@@ -0,0 +1,59 @@
+package syncclient
+
+import (
+	"bytes"
+	"context"
+	"github.com/gravitational/trace"
+	"sync"
+)
+
+type Cache struct {
```

Comment:
tl;dr:
- the cache is here to avoid rebuilding the client for each call
- the syncclient is here to keep track of who is using the client so we close it only once no one is using it.

As we don't support proper client reloading yet, the design is the following:
- tbot renews its certificate
- when accessing a client, we hit the cache
- the cache compares the current client certificates with the ones currently in tbot's memory and builds a new one if needed
- syncclient is a wrapper around a client that keeps track of how many users still hold a reference to the client (to avoid a client teardown if someone can still use it).

The name syncclient is not accurate. A saner description would be "usage-tracking-client". I'll gladly take any naming suggestion :)

Reply:
Still have some concerns about so many wrappers around that  but I don't have a better solution

Reply:
Fixed in https://github.com/gravitational/teleport/pull/34482